core/src/Streamly/Internal/FileSystem/Posix/ReadDir.hsc

```diff
@@ -164,7 +164,7 @@ lstatDname parent dname = do
 checkDirStatus
     :: PosixPath -> Ptr CChar -> #{type unsigned char} -> IO (Bool, Bool)
 checkDirStatus parent dname dtype =
-    if dtype == #const DT_UNKNOWN
+    if True
```

Comment:
Can we put this under a CPP macro, so that we can just enable/disable that for testing?